### PR TITLE
Latest updates for ODF module (install template and uninstall code)

### PR DESCRIPTION
**Pull request by kosilva93**

### CODE REVIEW

File: examples/odf/README.md [hunk 1]
```diff
@@ -8,38 +8,31 @@ You can define the IBM Cloud credentials in the IBM provider block but it is rec
...
-
-// Cluster parameters
-kube_config_path        = ".kube/config"
-worker_nodes            = 2  // Number of workers
-
-// ODF parameters
-resource_group_name     = "default"
-region                  = "us-east"
-cluster_id              = "<cluster-id>"
+cluster                 = "<cluster-id>"
```
Comment (annumberhocker) [inline examples/odf/README.md:28]: Do you need to list examples of the additional parameters?

File: examples/odf/main.tf [hunk 2]
```diff
@@ -1,17 +1,11 @@
...
-  source = "./../../modules/odf"
-  // TODO: With Terraform 0.13 replace the parameter 'enable' or the conditional expression using 'with_iaf' with 'count'
-  enable       = var.enable
-  cluster_id   = var.cluster_id
-  roks_version = var.roks_version
-
-  // Cluster parameters
-  //kube_config_path = data.ibm_container_cluster_config.cluster_config.config_file_path
-  kube_config_path = var.kube_config_path
+  source = "./.."
```
Comment (annumberhocker) [inline examples/odf/main.tf:6]: Is this correct?   Will this pick up the module code?  Shouldn't it still be `source = "./../../modules/odf"`

File: modules/odf/README.md [hunk 3]
```diff
@@ -2,38 +2,23 @@
 
 This Terraform Module installs the **ODF Service** on an Openshift (ROKS) cluster on IBM Cloud.
 
-**Module Source**: `github.com/ibm-hcbt/terraform-ibm-cloud-pak.git//modules/odf`
+A VPC cluster is required with at least three worker nodes. Each worker node must have a minimum of 16 CPUs and 64 GB RAM. https://cloud.ibm.com/docs/openshift?topic=openshift-deploy-odf-vpc for more information.
```
Comment (annumberhocker) [inline modules/odf/README.md:5]: VPC not required

File: modules/odf/main.tf [hunk 4]
```diff
@@ -2,16 +2,48 @@
 # Install ODF on the cluster
 ##################################
 
+locals {
+  installation_content = templatefile("${path.module}/templates/install_odf.yaml.tmpl", {
+    #monSize = var.monSize,
+    #monStorageClassName = var.monStorageClassName,
```
Comment (annumberhocker) [inline modules/odf/main.tf:8]: Need logic to check for version.  If 4.7, include `monSize` and `monStorageClassName`

File: modules/odf/scripts/install_odf.sh [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+#    echo "Supported version";
+    # ibmcloud ks cluster addon enable openshift-data-foundation -c ${CLUSTER} --version ${ROKS_VERSION} --param "odfDeploy=false"
+#    echo "Creating the ODF Customer Resource ${ODF_CR_CONTENT}"
+#kubectl apply -f -<<EOF
+#${ODF_CR_CONTENT}
+#EOF
+#else
+#    echo "Not supported version. Openshift version needs to be 4.7 or higher"; 
+#    exit 1
+#fi
```
Comment (annumberhocker) [inline modules/odf/scripts/install_odf.sh:25]: Remove this code if no longer needed

File: modules/odf/templates/install_odf.yaml.tmpl [hunk 6]
```diff
@@ -0,0 +1,11 @@
+apiVersion: ocs.ibm.io/v1
```
Comment (annumberhocker) [inline modules/odf/templates/install_odf.yaml.tmpl:1]: Need some sort of check between 4.7 and 4.8 to include the `monSize`, etc

**Review by annumberhocker — CHANGES_REQUESTED** [review]
Need a couple of changes and fix merge conflicts

File: examples/odf/README.md (hunk 1 above)
Comment (kosilva93) [reply examples/odf/README.md]: added

File: modules/odf/scripts/install_odf.sh (hunk 5 above)
Comment (kosilva93) [reply modules/odf/scripts/install_odf.sh:25]: done

File: examples/odf/main.tf (hunk 2 above)
Comment (kosilva93) [reply examples/odf/main.tf:6]: done

File: modules/odf/README.md (hunk 3 above)
Comment (kosilva93) [reply modules/odf/README.md:5]: done

File: examples/odf/vars.tf [hunk 7]
```diff
@@ -1,12 +1,60 @@
...
+  default = false
+}
+
+variable "clusterEncryption" {
+  description = "Enable encryption of storage cluster"
+  type = bool
+  default = false
+}
+
+# Options required for Openshift 4.7 only
```
Comment (annumberhocker) [inline examples/odf/vars.tf:49]: Add that this is only used for ROKS 4.7 in the description

File: modules/odf/variables.tf [hunk 8]
```diff
@@ -1,11 +1,62 @@
...
+}
+
+variable "clusterEncryption" {
+  description = "Enable encryption of storage cluster"
+  type = bool
+  default = false
+}
+
+# Options required for Openshift 4.7 only
+variable "monSize" {
```
Comment (annumberhocker) [inline modules/odf/variables.tf:52]: Add that this is required for 4.7 only to the description.

**Review by annumberhocker — APPROVED** [review]
LGTM, just a few changes to the descriptions in variables.tf files for monSize and monStorageClassName.

**Review by annumberhocker — APPROVED** [review]

**Pull request merged**